added FAQ entry on digital wallets as IdPs (#25)
usnistgov/800-63-FAQ: _63C/digitalwallet.md

---
question: C9
title: Can a digital identity wallet be an IdP?
---

> Digital identity wallets do not normally fulfill the requirements to be considered an IdP by the Digital Identity Guidelines. There has been interest in recent years in using applications to store cryptographically protected material on a user-carried device, a practice sometimes referred to as a digital identity wallet.  In their most common applications, digital identity wallets act as queryable personal data stores, collecting attributes from a variety of sources and presenting these attributes directly to relying parties.

Reviewer: ```suggestion
Digital identity wallets do not normally fulfill the requirements to be considered an IdP by the Digital Identity Guidelines.

There has been interest in recent years in using applications to store cryptographically protected material on a user-carried device, a practice sometimes referred to as a digital identity wallet.  In their most common applications, digital identity wallets act as queryable personal data stores, collecting attributes from a variety of sources and presenting these attributes directly to relying parties.
```

In most distributed scenarios that digital identity wallets are applied to, the RP's trust in the attributes tends to come from the ultimate sources of the attributes, not from the wallet itself. This is in contrast to the IdP model presented in the Guidelines, where the RP's trust in the attributes stems from its trust in the IdP asserting those attributes in a federation transaction. From the perspective of the Guidelines, the IdP is the source of the attributes. The IAL and AAL for a given federation transaction are applied to the entire account being presented, with any attributes asserted as being a part of that account.

> Furthermore, while the IdP model in the Guidelines does allow for presentation of attributes through an identity API in addition to being present in the assertion itself (see [Q-C8](#q-c8) for details), even in these cases, the identity API is assumed to be under the control of the IdP and inherits its trust from the IdP. Advanced use cases involving independent attribute providers, attribute metadata (including provenance), and distributed attribute models are outside the scope of the Guidelines.

Reviewer: ```suggestion
Furthermore, while the IdP model in the Guidelines does allow for presentation of attributes through an identity API in addition to being present in the assertion itself (see [Q-C8](#q-c8) for details), even in these cases, the identity API is assumed to be under the control of the IdP and inherits its trust from the IdP. Use cases involving independent attribute providers, attribute metadata (including provenance), and distributed attribute models are outside the scope of the Guidelines.
```
The issue isn't that we don't consider "advanced" use cases, but the difficulty of scaling trust to one IdP per subscriber.

Author: The one IDP per person isn't the focus of this sentence, but I'm fine with pulling back on the adjective.

> In the end, while it's technically feasible for a wallet to act as an IdP, it would need to truly act in the role of an IdP including having the RP's connecting to the wallet trust the wallet's ability to present assertions and attributes.

Reviewer: ```suggestion
In the end, while it's technically possible for a wallet to act as an IdP, to be useful in practice it would need to truly act in that role. This includes the need for RPs to establish trust for assertions and attributes coming from the wallet.
```

Author: ```suggestion
In the end, while it's technically possible for a wallet to act as an IdP, to be useful in practice it would need to truly act in that role. This includes the need for RPs to establish trust for assertions and attributes coming from the wallet as well as the wallet's ability to present these assertions and attributes to the RP.
```

Author: Didn't want to lose the presentation aspect to this, but the rest is fine. Incorporating both together, please take a look.

---
question: C9
title: Can a digital wallet be an IdP?
---

> While OpenID Connect and SAML are the most common federated identity protocols in use today, the Digital Identity Guidelines allow any protocol that meets the requirements outlined in the system to act as a federation protocol. Similarly, while a web-hosted server is the most common deployment for an IdP, the Digital Identity Guidelines do not assume any particular deployment model for IdPs. 

Reviewer: ```suggestion
While OpenID Connect and SAML are the most common federated identity protocols in use today, the Digital Identity Guidelines allow any protocol that meets the requirements outlined in the Guidelines to act as a federation protocol. Similarly, while a web-hosted server is the most common deployment for an IdP, the Digital Identity Guidelines do not assume any particular deployment model for IdPs. 
```

> There has been interest in recent years in using applications to store cryptographically protected material on a user-carried device, a practice sometimes referred to as a digital wallet. Given that the definition of IdP does not assume a publicly deployed web service, a digital wallet could serve the function of an IdP as long as it fulfills all the requirements laid out in the guidelines. Namely, the wallet would need to create a signed assertion (which still needs to meet all the general assertion requirements) and somehow transfer that assertion to the RP over a network using either a front-channel or back-channel presentation mechanism. The RP in turn would need to be able to validate the assertion's signature and contents, as well as determine that the IdP is a trustworthy party for generating the assertion. There are numerous methods for establishing an IdP-RP trust relationship discussed in the Digital Identity Guidelines, and any of these could be applicable to trusting assertions from a digital wallet style application.

Reviewer: I presume that the wallet itself needs to do the signing in order to make sure that the assertion has the correct audience, etc. Is that correct, or does it sign a pre-signed assertion (which I suppose could be encapsulated in another assertion with audience)? In any case, trust for a wallet held by an individual subscriber seems like a scaling challenge.

Author: Yes, the wallet would need to sign the assertion, audience restrict it, and do all of those other things that are spelled out in the guidelines as requirements for assertions. Usually a wallet is acting more like an attribute provider so it would need to go an extra step of having an assertion package to wrap them in.

Trust is definitely a challenge in the space, but that's not the challenge we're trying to solve here. It's akin to having a personal IDP and convincing RPs to accept it -- technologically possible, and past that it's up to the RP.

Author: If we're capitalizing Guidelines this is another one:

```suggestion
There has been interest in recent years in using applications to store cryptographically protected material on a user-carried device, a practice sometimes referred to as a digital wallet. Given that the definition of IdP does not assume a publicly deployed web service, a digital wallet could serve the function of an IdP as long as it fulfills all the requirements laid out in the Guidelines. Namely, the wallet would need to create a signed assertion (which still needs to meet all the general assertion requirements) and somehow transfer that assertion to the RP over a network using either a front-channel or back-channel presentation mechanism. The RP in turn would need to be able to validate the assertion's signature and contents, as well as determine that the IdP is a trustworthy party for generating the assertion. There are numerous methods for establishing an IdP-RP trust relationship discussed in the Digital Identity Guidelines, and any of these could be applicable to trusting assertions from a digital wallet style application.
```

Reviewer 2: Regarding the requirement to sign the assertion and related attribute information, what entity is the signing authority? I have looked at this as being self-asserted information -- so if the wallet owner is the signing authority then this is just signed self-asserted information. The principle in 800-63 is that self-asserted information does not bear trust. This is the question that I want to discuss on this FAQ.
